app/models/concerns/publishable.rb

```diff
@@ -20,6 +20,10 @@ module Publishable
       where condition
     }
 
+    before_validation -> {
+      self.published_on ||= Time.zone.now if published?
```

Comment:
Why Time.zone.now instead of Time.now? If it writes to DB I think one is safe to rely on Time.now to return the UTC time (which we use internally). Isn't Time.zone.now only for presentation?